### PR TITLE
[SHELL32] Implement 'New Link' (Retrial of #1510)

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
This PR is retrial of #1510.
Right-Click Desktop, and Choose [New] --> [New Link]. Wizard will start up.
JIRA issue: [CORE-15511](https://jira.reactos.org/browse/CORE-15511)
Movie: https://jira.reactos.org/secure/attachment/52382/Shortcut2ndTrial.webm
## Proposed changes
- Set `NULL` to m_pItems in `CNewMenu::UnloadAllItems()`.
- Add `bIsShortcut` member into `SHELLNEW_ITEM` structure.
- Correctly create cache about `".lnk"`.
- Add special handling for the item of `bIsShortcut == TRUE`.
- Add `bRename` parameter to `CNewMenu::SelectNewItem`.
- Add and use `CNewMenu::NewItemByCommand` and `CNewMenu::NewItemByNonCommand` methods.

### CODE REVIEW

File: dll/win32/shell32/CNewMenu.cpp [hunk 1]
```diff
@@ -463,121 +463,145 @@ HRESULT CNewMenu::CreateNewFolder(LPCMINVOKECOMMANDINFO lpici)
...
-HRESULT CNewMenu::CreateNewItem(SHELLNEW_ITEM *pItem, LPCMINVOKECOMMANDINFO lpcmi)
+HRESULT CNewMenu::NewItemByCommand(SHELLNEW_ITEM *pItem, LPCWSTR wszPath)
+{
+    WCHAR wszBuf[MAX_PATH], wszShortPath[MAX_PATH];
+    LPWSTR Ptr, pwszCmd;
+    WCHAR wszTemp[MAX_PATH];
+    STARTUPINFOW si;
+    PROCESS_INFORMATION pi;
+
+    if (!GetShortPathNameW(wszPath, wszShortPath, _countof(wszShortPath)))
```
Comment (HBelusca) [inline dll/win32/shell32/CNewMenu.cpp]: This is going to be buggy in general.

Comment (katahiromz) [reply dll/win32/shell32/CNewMenu.cpp]: `rundll32` requires short path name.

Comment (HBelusca) [reply dll/win32/shell32/CNewMenu.cpp]: I'm on it.

Comment (HBelusca) [reply dll/win32/shell32/CNewMenu.cpp]: I've tested, and it works fine with long path names with or without spaces.

**Pull request merged**